src/opm/output/eclipse/Summary.cpp

```diff
@@ -2642,6 +2673,145 @@ namespace Evaluator {
         }
     };
 
+    class InterRegionValue : public Base
+    {
+    public:
+        explicit InterRegionValue(Opm::EclIO::SummaryNode node,
+                                  const Opm::UnitSystem::measure m)
+            : node_   (std::move(node))
```

Comment:
Why use a by-value parameter just to move it afterwards? Won't there be copy just like when passing by const-reference and then doing the copy. The latter just seems less complicate/fancy and to me clearer, What am I missing?

Reply:
> Why use a by-value parameter just to move it afterwards?

It's the "best practice" way of passing constructor parameters that will be copied into the object, since it enables move semantics for temporaries.  In this case it probably doesn't matter too much since `std::move()` on a `SummaryNode` is mostly copy anyway except for the data members of standard library types.

> Won't there be copy just like when passing by const-reference and then doing the copy.

Yes.

Reply:
Would you mind giving a reference for this "best practice".

Reply:
> Would you mind giving a reference for this "best practice". I mean using std:move always.

Sorry for being unclear.  I did not mean to say that we should **always** write `std::move()`.  I was talking specifically about the case of transferring ownership of an existing object into a larger container.  Herb Sutter gave a plenary talk at [CppCon 2014](https://www.youtube.com/watch?v=xnqTKD8uD64&ab_channel=CppCon) about this topic and others, and his discussion about this starts around the 1 hour mark.  I recommend watching the whole 1 hour and 40 minutes talk if you have the time.